src/passes/ReorderGlobals.cpp

```diff
@@ -237,14 +236,12 @@ struct ReorderGlobals : public Pass {
     }
 
     // Apply the indices we computed.
-    std::sort(
-      globals.begin(),
-      globals.end(),
-      [&](const std::unique_ptr<Global>& a, const std::unique_ptr<Global>& b) {
-        return (*best)[originalIndices[a->name]] <
-               (*best)[originalIndices[b->name]];
-      });
-
+    std::vector<std::unique_ptr<Global>> old;
+    old.swap(globals);
```

Comment:
```suggestion
    std::vector<std::unique_ptr<Global>> old = std::move(globals);
```

Reply:
Hmm, it does appear that move assignment for vectors uses the "valid but unspecified" language. But their move constructor [further guarantees](https://en.cppreference.com/w/cpp/container/vector/vector) that the operand is left in the empty state.
```suggestion
    std::vector<std::unique_ptr<Global>> old(std::move(globals));
```

Reply:
Interesting. I guess specific cases are ok then, even if the general rule disallows it. But I think it's safer not to assume the specific case here, as we may replace the `std::vector` with some other data structure some day.

Reply:
Right, `std::move` itself imposes no requirements except that the result be "valid." Individual types further specify what happens when they are moved. All container types should do the right thing here, so I think the extra code clarity is worth it.

Reply:
Fair enough, I guess we would do the same in any custom data structures we use too.